packages/flutter_tools/lib/src/cache.dart

```diff
@@ -210,13 +224,26 @@ class Cache {
     return cachedFile.path;
   }
 
-  Future<void> updateAll() async {
-    if (!_lockEnabled)
+  Future<void> updateAll({
+    List<BuildMode> buildModes = const <BuildMode>[],
+    List<TargetPlatform> targetPlatforms = const <TargetPlatform>[],
+    bool skipUnknown = true,
+    bool clobber = false,
+  }) async {
+    if (!_lockEnabled) {
       return;
+    }
     try {
       for (CachedArtifact artifact in _artifacts) {
-        if (!artifact.isUpToDate())
-          await artifact.update();
+        bool localClobber = clobber;
+        if (localClobber) {
+          await artifact.update(buildModes: buildModes, targetPlatforms: targetPlatforms, skipUnknown: skipUnknown, clobber: localClobber);
```

Comment:
I know we're pretty lax about line length, but these `update` calls are getting a little long to read.

Reply:
Since the tools don't contain widget trees, it seems like we should really consider dartfmt